crates/ty_python_semantic/src/types/class.rs

```diff
@@ -173,6 +173,10 @@ impl<'db> GenericAlias<'db> {
         Self::new(db, self.origin(db), self.specialization(db).normalized(db))
     }
 
+    pub(super) fn materialize(self, db: &'db dyn Db) -> Self {
+        Self::new(db, self.origin(db), self.specialization(db).materialize(db))
```

Comment:
I think any time we have a `materialize` method that doesn't accept `variance` as a parameter, it means we have a bug where we won't handle nested types correctly. And I think this is no exception. (I'll comment in more detail below.)